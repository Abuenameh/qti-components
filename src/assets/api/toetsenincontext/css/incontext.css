.inlinechoicetext {
  display: inline-block;
  color: var(--qti-text-color);
  border-radius: var(--qti-border-radius);
  background: var(--qti-background);
  box-shadow: inset 0px var(--qti-border-width) var(--qti-border-color);
}
.inlinechoicetext:empty {
  width: 50%;
}
.inlinechoicetext:empty:after {
  content: '\00a0';
}

qti-hottext {
  display: unset !important;
}
qti-hottext:before,
qti-hottext:after {
  content: unset !important;
}

qti-hottext[aria-checked="true"] {
  text-decoration: underline;
}

/* [qti-hottext-interaction] [qti-choice] {
  display: unset !important;
} */
