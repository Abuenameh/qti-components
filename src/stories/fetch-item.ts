import { qtiTransformItem, qtiTransformTest } from 'src/lib/qti-transformers';

export const fetchItem = async (packageUri: string, index: number): Promise<any> => {
  packageUri = packageUri.endsWith('/') ? packageUri : packageUri + '/';
  const assessmentTestHref = `assessment.xml`;
  const itemLocation = `${packageUri}${assessmentTestHref.substring(0, assessmentTestHref.lastIndexOf('/'))}/`;

  const itemsFromTest = await qtiTransformTest()
    .load(packageUri + assessmentTestHref)
    .then(api => api.items());

  const itemHTML = await qtiTransformItem()
    .load(itemLocation + itemsFromTest[index].href)
    .then(api => api.path(itemLocation).html());

  return { itemXML: itemHTML, items: itemsFromTest };
};
