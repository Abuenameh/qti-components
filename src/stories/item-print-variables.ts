import { Signal } from '@lit-labs/preact-signals';
import { consume } from '@lit/context';
import { LitElement, html } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { ItemContext, itemContext } from '../lib/qti-components/qti-assessment-item/qti-item.context';

@customElement('item-print-variables')
export class ItemPrintVariables extends LitElement {
  @consume({ context: itemContext })
  @state()
  public itemContext?: Signal<ItemContext>;

  render() {
    return html` <pre>${JSON.stringify(this.itemContext, null, 2)}</pre> `;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'item-print-variables': ItemPrintVariables;
  }
}
