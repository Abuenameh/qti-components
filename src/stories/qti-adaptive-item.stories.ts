import { html } from 'lit';
import { createRef, ref } from 'lit-html/directives/ref.js';

import { action } from '@storybook/addon-actions';

import '../lib/qti-components/index';
import { QtiAssessmentItem } from '../lib/qti-components/qti-assessment-item/qti-assessment-item';

export default {
  component: 'qti-adaptive-item'
};

export const Printed = () => {
  const assessmentItemRef = createRef<QtiAssessmentItem>();

  return html`<qti-assessment-item
    identifier="ITEM-ZONNEPANNELEN"
    ${ref(assessmentItemRef)}
    @qti-outcome-changed=${action(`qti-outcome-changed`)}
    @qti-interaction-changed=${action(`qti-interaction-changed`)}
  >
    <qti-response-declaration identifier="RESPONSE" cardinality="single" base-type="integer"></qti-response-declaration>

    <qti-response-declaration identifier="RESPONSE2" cardinality="single" base-type="identifier">
      <qti-correct-response>
        <qti-value>MINDER</qti-value>
      </qti-correct-response>
    </qti-response-declaration>

    <qti-response-declaration
      identifier="RESPONSE3"
      cardinality="single"
      base-type="integer"
    ></qti-response-declaration>

    <qti-outcome-declaration identifier="STORY" cardinality="single" base-type="identifier">
      <qti-default-value>
        <qti-value>ZONNEPANELEN</qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-outcome-declaration identifier="SCORE" cardinality="single" base-type="float" normal-maximum="10.0">
      <qti-default-value>
        <qti-value>0</qti-value>
      </qti-default-value>
    </qti-outcome-declaration>

    <qti-item-body>
      <qti-feedback-block outcome-identifier="STORY" show-hide="show" identifier="ZONNEPANELEN">
        <p>Hoeveel zonnepannelen zijn er</p>
        <qti-text-entry-interaction response-identifier="RESPONSE"></qti-text-entry-interaction>
      </qti-feedback-block>

      <qti-feedback-block outcome-identifier="STORY" show-hide="show" identifier="UITSTOOT">
        <p>
          Zonnepanelen zorgen voor
          <qti-inline-choice-interaction response-identifier="RESPONSE2">
            <qti-inline-choice identifier="GEEN">geen</qti-inline-choice>
            <qti-inline-choice identifier="MINDER">minder</qti-inline-choice>
            <qti-inline-choice identifier="EVENVEEL">evenveel</qti-inline-choice>
            <qti-inline-choice identifier="MEER">meer</qti-inline-choice>
          </qti-inline-choice-interaction>
          CO2 uitstoot dan het gebruik van fossiele branstoffen.
        </p>
      </qti-feedback-block>

      <qti-feedback-block outcome-identifier="STORY" show-hide="show" identifier="HOEVEEL">
        <p>Hoeveel = <qti-printed-variable identifier="RESPONSE"></qti-printed-variable> x 5</p>
        <qti-text-entry-interaction response-identifier="RESPONSE3"></qti-text-entry-interaction>
      </qti-feedback-block>
      <qti-end-attempt-interaction title="End attempt"></qti-end-attempt-interaction>
    </qti-item-body>

    <qti-response-processing>
      <qti-set-outcome-value identifier="STORY">
        <qti-base-value base-type="identifier">UITSTOOT</qti-base-value>
      </qti-set-outcome-value>

      <qti-response-condition>
        <qti-response-if>
          <qti-not>
            <qti-is-null>
              <qti-variable identifier="RESPONSE2"></qti-variable>
            </qti-is-null>
          </qti-not>
          <qti-response-condition>
            <qti-response-if>
              <qti-match>
                <qti-variable identifier="RESPONSE2"></qti-variable>
                <qti-correct identifier="RESPONSE2"></qti-correct>
              </qti-match>
              <qti-set-outcome-value identifier="SCORE">
                <qti-base-value base-type="float">1</qti-base-value>
              </qti-set-outcome-value>
              <qti-set-outcome-value identifier="STORY">
                <qti-base-value base-type="identifier">HOEVEEL</qti-base-value>
              </qti-set-outcome-value>
            </qti-response-if>
            <qti-response-else>
              <qti-set-outcome-value identifier="SCORE">
                <qti-base-value base-type="float">0</qti-base-value>
              </qti-set-outcome-value>
            </qti-response-else>
          </qti-response-condition>
        </qti-response-if>

        <qti-response-else>
          <qti-response-condition>
            <qti-response-if>
              <qti-not>
                <qti-is-null>
                  <qti-variable identifier="RESPONSE3"></qti-variable>
                </qti-is-null>
              </qti-not>
              <qti-response-condition>
                <qti-response-if>
                  <qti-equal>
                    <qti-product>
                      <qti-base-value base-type="integer">5</qti-base-value>
                      <qti-variable identifier="RESPONSE"></qti-variable>
                    </qti-product>
                    <qti-variable identifier="RESPONSE3"></qti-variable>
                  </qti-equal>
                  <qti-set-outcome-value identifier="SCORE">
                    <qti-base-value base-type="float">1</qti-base-value>
                  </qti-set-outcome-value>
                </qti-response-if>
                <qti-response-else>
                  <qti-set-outcome-value identifier="SCORE">
                    <qti-base-value base-type="float">0</qti-base-value>
                  </qti-set-outcome-value>
                </qti-response-else>
              </qti-response-condition>
            </qti-response-if>
          </qti-response-condition>
        </qti-response-else>
      </qti-response-condition>
    </qti-response-processing>
  </qti-assessment-item>`;
};
