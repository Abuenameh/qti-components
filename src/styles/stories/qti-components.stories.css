/* TEST TEST TEST TEMPLATE, just for storybook */
/* IS NOT EXPORTED IN ANY WAY TO THE LIBRARY */

@import 'tailwindcss/base';

@import 'tailwindcss/components';
@import '../qti-components.css';

@import 'tailwindcss/utilities';

@config "./tailwind.config.qti-components.cjs";
