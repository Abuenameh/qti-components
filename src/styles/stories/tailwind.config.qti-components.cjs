/** @type {import('tailwindcss').Config} */
/* eslint-env node */

/* 
This config is only used to preview the theme in the story
qti-theme.css. This css file imports this configuration '@config "../../tailwind.config.theme.cjs"'
and watches the classes used in the qti-theme.stories.ts for developing the UI components
*/

const tailwindconfig = require('../../../tailwind.config.cjs')

/* PK: extend the default config with content of the themes storie */
export default {
  ...tailwindconfig,
  content: ['src/styles/stories/qti-components.stories.ts']
}
