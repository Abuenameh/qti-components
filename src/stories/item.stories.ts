import '@citolab/qti-components/qti-components';

import { html } from 'lit';
import { action } from '@storybook/addon-actions';
import { unsafeHTML } from 'lit/directives/unsafe-html.js';
import type { Meta, StoryObj } from '@storybook/web-components';
import { fetchItem } from './fetch-item';
import packages from '../assets/packages.json';

const meta: Meta = {
  title: 'items',
  argTypes: {
    view: {
      options: ['candidate', 'scorer'],
      control: { type: 'radio' }
    },
    qtipkg: {
      options: packages.packages,
      control: { type: 'radio' }
    },
    disabled: { control: { type: 'boolean' } }
  },
  args: {
    serverLocation: 'http://localhost:6006/api',
    qtipkg: 'biologie',
    itemIndex: 0
  }
};

export default meta;
type Story = StoryObj;

let item;

export const Items: Story = {
  render: ({ disabled, view }, { argTypes, loaded: { xml } }) => {
    const onInteractionChangedAction = action('on-interaction-changed');
    const onOutcomeChangedAction = action('qti-outcome-changed');
    const onItemFirstUpdated = ({ detail: qtiAssessmentItem }) => {
      item = qtiAssessmentItem;
      action('qti-item-first-updated');
    };

    item && (item.disabled = disabled);
    item && (item.view = view);

    return html`
      <div
        @qti-interaction-changed=${onInteractionChangedAction}
        @qti-outcome-changed=${onOutcomeChangedAction}
        @qti-item-first-updated=${onItemFirstUpdated}
      >
        ${unsafeHTML(xml.itemXML)}
      </div>
    `;
  },
  loaders: [async ({ args }) => ({ xml: await fetchItem(`${args.serverLocation}/${args.qtipkg}`, args.itemIndex) })]
};
