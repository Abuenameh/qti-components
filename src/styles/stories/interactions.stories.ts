import { html } from 'lit';
import { Interaction as Choice } from '../../lib/qti-components/qti-interaction/qti-choice-interaction/qti-choice-interaction.stories';
import { Interaction as TextEntry } from '../../lib/qti-components/qti-interaction/qti-text-entry-interaction/qti-text-entry-interaction.stories';
import { Interaction as ExtendedText } from '../../lib/qti-components/qti-interaction/qti-extended-text-interaction/qti-extended-text-interaction.stories';
import { Interaction as GapMatch } from '../../lib/qti-components/qti-interaction/qti-gap-match-interaction/qti-gap-match-interaction.stories';
import { Poly as HotSpot } from '../../lib/qti-components/qti-interaction/qti-hotspot-interaction/qti-hotspot-interaction.stories';
import { checkbox as Hottext } from '../../lib/qti-components/qti-interaction/qti-hottext-interaction/qti-hottext-interaction.stories';
import { Interaction as InlineChoice } from '../../lib/qti-components/qti-interaction/qti-inline-choice-interaction/qti-inline-choice-interaction.stories';
import { Interaction as Match } from '../../lib/qti-components/qti-interaction/qti-match-interaction/qti-match-interaction.stories';
import { Interaction as Order } from '../../lib/qti-components/qti-interaction/qti-order-interaction/qti-order-interaction.stories';
import { Interaction as GraphicOrder } from '../../lib/qti-components/qti-interaction/qti-graphic-order-interaction/qti-graphic-order-interaction.stories';
import { Interaction as Associate } from '../../lib/qti-components/qti-interaction/qti-associate-interaction/qti-associate-interaction.stories';
import { Interaction as GraphicAssociate } from '../../lib/qti-components/qti-interaction/qti-graphic-associate-interaction/qti-graphic-associate-interaction.stories';
import { Interaction as GraphicGapMatch } from '../../lib/qti-components/qti-interaction/qti-graphic-gap-match-interaction/qti-graphic-gap-match-interaction.stories';
import { Interaction as Media } from '../../lib/qti-components/qti-interaction/qti-media-interaction/qti-media-interaction.stories';
import { Interaction as PositionObject } from '../../lib/qti-components/qti-interaction/qti-position-object-interaction/qti-position-object-interaction.stories';
import { Interaction as SelectPoint } from '../../lib/qti-components/qti-interaction/qti-select-point-interaction/qti-select-point-interaction.stories';
import { Interaction as Slider } from '../../lib/qti-components/qti-interaction/qti-slider-interaction/qti-slider-interaction.stories';
// import { Interaction as Upload } from '../lib/qti-components/qti-interaction/qti-upload-interaction/qti-upload-interaction.stories';
// import { Interaction as Drawing } from '../lib/qti-components/qti-interaction/qti-drawing-interaction/qti-drawing-interaction.stories';
// import { Interaction as EndAttempt } from '../lib/qti-components/qti-interaction/qti-end-attempt-interaction/qti-end-attempt-interaction.stories';
// import { Interaction as Custom } from '../lib/qti-components/qti-interaction/qti-custom-interaction/qti-custom-interaction.stories';
import { Interaction as PortableCustom } from '../../lib/qti-components/qti-interaction/qti-portable-custom-interaction/qti-portable-custom-interaction.stories';

export { Choice };

export default {
  title: 'styles/themes'
};
export const Interactions = args => html`
  <!-- <qti-item-body> -->
  <div style="display:flex; gap:0.5rem">
    <!-- <h2>gap-gap-match-interaction</h2> -->
    ${GapMatch.render({})}
    <!-- <h2>qti-match-interaction</h2> -->
    ${Match.render({})}
  </div>
  <div style="display:flex; gap:0.5rem">
    <!-- <h2>qti-order-interaction</h2> -->
    ${Order.render({})}
    <!-- <h2>qti-associate-interaction</h2> -->
    ${Associate.render({})}
  </div>
  <h2>qti-graphic-gap-match-interaction</h2>
  ${GraphicGapMatch.render({})}

  <h2>choice-interaction</h2>
  <qti-layout-row>
    <qti-layout-col6>${Choice.render({})}</qti-layout-col6>
    <qti-layout-col6> ${Choice.render({ disabled: true, 'max-choices': 2 })}</qti-layout-col6>
  </qti-layout-row>
  ${Choice.render({
    orientation: 'vertical',
    classes: ['qti-input-control-hidden', 'qti-choices-stacking-2']
  })}

  <h2>text-entry-interaction</h2>
  ${TextEntry.render({})}
  <h2>extended-text-interaction</h2>
  ${ExtendedText.render({})}

  <h2>qti-hotspot-interaction</h2>
  ${HotSpot.render()}
  <h2>qti-hottext-interaction</h2>
  ${Hottext.render({ 'max-choices': '1' })} ${Hottext.render({ 'max-choices': '2' })}
  ${Hottext.render({ classes: ['qti-input-control-hidden'] })}
  <h2>qti-inline-choice-interaction</h2>
  ${InlineChoice.render({})}
  <h2>qti-graphic-order-interaction</h2>
  ${GraphicOrder.render({})}
  <h2>qti-graphic-associate-interaction</h2>
  ${GraphicAssociate.render({})}
  <h2>qti-media-interaction</h2>
  ${Media.render({})}
  <h2>qti-position-object-interaction</h2>
  ${PositionObject.render({})}
  <h2>qti-select-point-interaction</h2>
  ${SelectPoint.render({})}
  <h2>qti-slider-interaction</h2>
  ${Slider.render({})}
  <!-- 
  <div>Upload.render({})</div>
  <div>Drawing.render({})</div>
  <div>EndAttempt.render({})</div>
  <div>Custom.render({})</div>
  -->

  <!-- <div>PortableCustom.render({})</div>
    -->
  <!-- </qti-item-body> -->
`;
