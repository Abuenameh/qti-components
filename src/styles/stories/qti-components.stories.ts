import { html } from 'lit';

import './qti-components.stories.css';

export default {
  title: 'styles/themes'
};

class ButtonComponent extends HTMLElement {
  shadow: ShadowRoot;
  constructor() {
    super();
    this.shadow = this.attachShadow({ mode: 'open' });
  }
  connectedCallback() {
    this.render();
  }
  render() {
    this.shadow.innerHTML = `<div part="ch"><div part="cha"></div></div><slot></slot><div part="drop"></div>`;
  }
}
window.customElements.define('button-component', ButtonComponent);

const components = (mod: string) => html`<div class="grid grid-cols-6 gap-4 p-4">
  <div class="text-xs">button</div>
  <div class="text-xs">hover</div>
  <div class="text-xs">focus</div>
  <div class="text-xs">active</div>
  <div class="text-xs">active+focus</div>
  <div class="text-xs">disabled</div>

  <button-component role="radio" class="check ${mod}" aria-checked="false">check</button-component>
  <button-component role="radio" class="check hov ${mod}" aria-checked="false">check</button-component>
  <button-component role="radio" class="check foc ${mod}" aria-checked="false">check</button-component>
  <button-component role="radio" class="check act ${mod}" aria-checked="true">check</button-component>
  <button-component role="radio" class="check foc act ${mod}" aria-checked="true">check</button-component>
  <button-component role="radio" class="check dis ${mod}" aria-checked="false">check</button-component>

  <button-component role="checkbox" class="check ${mod}" aria-checked="false">check</button-component>
  <button-component role="checkbox" class="check hov ${mod}" aria-checked="false">check</button-component>
  <button-component role="checkbox" class="check foc ${mod}" aria-checked="false">check</button-component>
  <button-component role="checkbox" class="check act ${mod}" aria-checked="true">check</button-component>
  <button-component role="checkbox" class="check foc act ${mod}" aria-checked="true">check</button-component>
  <button-component role="checkbox" class="check dis ${mod}" aria-checked="false">check</button-component>

  <div class="button ${mod}">button</div>
  <div class="button ${mod} hov ">button</div>
  <div class="button ${mod} foc ">button</div>
  <div class="button ${mod} act ">button</div>
  <div class="button ${mod} foc act ">button</div>
  <div class="button ${mod} dis ">button</div>

  <div class="spot ${mod}"></div>
  <div class="spot ${mod} hov "></div>
  <div class="spot ${mod} foc "></div>
  <div class="spot ${mod} act "></div>
  <div class="spot ${mod} foc act "></div>
  <div class="spot ${mod} dis "></div>

  <div class="drag ${mod}">drag</div>
  <div class="drag hov ${mod}">drag</div>
  <div class="drag foc ${mod}">drag</div>
  <div class="drag act ${mod}">drag</div>
  <div class="drag foc act ${mod}">drag</div>
  <div class="drag dis ${mod}">drag</div>

  <div class="drop ${mod}"></div>
  <div class="drop hov ${mod}"></div>
  <div class="drop foc ${mod}"></div>
  <div class="drop act ${mod}"></div>
  <div class="drop foc act ${mod}"></div>
  <div class="drop dis ${mod}"></div>

  <div class="drop ${mod}"><div class="drag ${mod}">drag</div></div>
  <div class="drop hov ${mod}"><div class="drag ${mod}">drag</div></div>
  <div class="drop foc ${mod}"><div class="drag ${mod}">drag</div></div>
  <div class="drop act ${mod}"><div class="drag ${mod}">drag</div></div>
  <div class="drop foc act ${mod}"><div class="drag ${mod}">drag</div></div>
  <div class="drop dis ${mod}"><div class="drag ${mod}">drag</div></div>

  <div class="drop ${mod}">
    With text
    <div class="drag ${mod}">drag</div>
  </div>
  <div class="drop hov ${mod}">
    With text
    <div class="drag ${mod}">drag</div>
  </div>
  <div class="drop foc ${mod}">
    With text
    <div class="drag ${mod}">drag</div>
  </div>
  <div class="drop act ${mod}">
    With text
    <div class="drag ${mod}">drag</div>
  </div>
  <div class="drop foc act ${mod}">
    With text
    <div class="drag ${mod}">drag</div>
  </div>
  <div class="drop dis ${mod}">
    With text
    <div class="drag ${mod}">drag</div>
  </div>

  <input type="text" class="text ${mod}" value="Button text" />
  <input type="text" class="text hov ${mod}" value="Button text" />
  <input type="text" class="text foc ${mod}" value="Button text" />
  <input type="text" class="text act ${mod}" value="Button text" />
  <input type="text" class="text foc act ${mod}" value="Button text" />
  <input type="text" class="text dis ${mod}" value="Button text" />

  <select class="select ${mod}">
    <option>Button text</option>
  </select>
  <select class="select hov ${mod}">
    <option>Button text</option>
  </select>
  <select class="select foc ${mod}">
    <option>Button text</option>
  </select>
  <select class="select act ${mod}">
    <option>Button text</option>
  </select>
  <select class="select foc act ${mod}">
    <option>Button text</option>
  </select>
  <select class="select dis ${mod}">
    <option>Button text</option>
  </select>
</div>`;

export const Components = args => html` ${components('')} ${components('sm')}`;

// export const Components = args => html`<button class="button">asasd</button> <button class="button-in">asasd</button>`;
