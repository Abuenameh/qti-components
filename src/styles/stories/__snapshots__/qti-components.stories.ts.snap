// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles/themes Components smoke-test 1`] = `
<div id="root-inner">
  <div class="grid grid-cols-6 gap-4 p-4">
    <div class="text-xs">
      button
    </div>
    <div class="text-xs">
      hover
    </div>
    <div class="text-xs">
      focus
    </div>
    <div class="text-xs">
      active
    </div>
    <div class="text-xs">
      active+focus
    </div>
    <div class="text-xs">
      disabled
    </div>
    <button-component role="radio"
                      aria-checked="false"
                      class="check "
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="false"
                      class="check hov "
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="false"
                      class="check foc "
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="true"
                      class="check act "
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="true"
                      class="check foc act "
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="false"
                      class="check dis "
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="false"
                      class="check "
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="false"
                      class="check hov "
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="false"
                      class="check foc "
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="true"
                      class="check act "
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="true"
                      class="check foc act "
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="false"
                      class="check dis "
    >
      check
    </button-component>
    <div class="button ">
      button
    </div>
    <div class="button  hov ">
      button
    </div>
    <div class="button  foc ">
      button
    </div>
    <div class="button  act ">
      button
    </div>
    <div class="button  foc act ">
      button
    </div>
    <div class="button  dis ">
      button
    </div>
    <div class="spot ">
    </div>
    <div class="spot  hov ">
    </div>
    <div class="spot  foc ">
    </div>
    <div class="spot  act ">
    </div>
    <div class="spot  foc act ">
    </div>
    <div class="spot  dis ">
    </div>
    <div class="drag ">
      drag
    </div>
    <div class="drag hov ">
      drag
    </div>
    <div class="drag foc ">
      drag
    </div>
    <div class="drag act ">
      drag
    </div>
    <div class="drag foc act ">
      drag
    </div>
    <div class="drag dis ">
      drag
    </div>
    <div class="drop ">
    </div>
    <div class="drop hov ">
    </div>
    <div class="drop foc ">
    </div>
    <div class="drop act ">
    </div>
    <div class="drop foc act ">
    </div>
    <div class="drop dis ">
    </div>
    <div class="drop ">
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop hov ">
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop foc ">
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop act ">
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop foc act ">
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop dis ">
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop ">
      With text
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop hov ">
      With text
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop foc ">
      With text
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop act ">
      With text
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop foc act ">
      With text
      <div class="drag ">
        drag
      </div>
    </div>
    <div class="drop dis ">
      With text
      <div class="drag ">
        drag
      </div>
    </div>
    <input type="text"
           value="Button text"
           class="text "
    >
    <input type="text"
           value="Button text"
           class="text hov "
    >
    <input type="text"
           value="Button text"
           class="text foc "
    >
    <input type="text"
           value="Button text"
           class="text act "
    >
    <input type="text"
           value="Button text"
           class="text foc act "
    >
    <input type="text"
           value="Button text"
           class="text dis "
    >
    <select class="select ">
      <option>
        Button text
      </option>
    </select>
    <select class="select hov ">
      <option>
        Button text
      </option>
    </select>
    <select class="select foc ">
      <option>
        Button text
      </option>
    </select>
    <select class="select act ">
      <option>
        Button text
      </option>
    </select>
    <select class="select foc act ">
      <option>
        Button text
      </option>
    </select>
    <select class="select dis ">
      <option>
        Button text
      </option>
    </select>
  </div>
  <div class="grid grid-cols-6 gap-4 p-4">
    <div class="text-xs">
      button
    </div>
    <div class="text-xs">
      hover
    </div>
    <div class="text-xs">
      focus
    </div>
    <div class="text-xs">
      active
    </div>
    <div class="text-xs">
      active+focus
    </div>
    <div class="text-xs">
      disabled
    </div>
    <button-component role="radio"
                      aria-checked="false"
                      class="check sm"
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="false"
                      class="check hov sm"
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="false"
                      class="check foc sm"
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="true"
                      class="check act sm"
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="true"
                      class="check foc act sm"
    >
      check
    </button-component>
    <button-component role="radio"
                      aria-checked="false"
                      class="check dis sm"
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="false"
                      class="check sm"
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="false"
                      class="check hov sm"
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="false"
                      class="check foc sm"
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="true"
                      class="check act sm"
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="true"
                      class="check foc act sm"
    >
      check
    </button-component>
    <button-component role="checkbox"
                      aria-checked="false"
                      class="check dis sm"
    >
      check
    </button-component>
    <div class="button sm">
      button
    </div>
    <div class="button sm hov ">
      button
    </div>
    <div class="button sm foc ">
      button
    </div>
    <div class="button sm act ">
      button
    </div>
    <div class="button sm foc act ">
      button
    </div>
    <div class="button sm dis ">
      button
    </div>
    <div class="spot sm">
    </div>
    <div class="spot sm hov ">
    </div>
    <div class="spot sm foc ">
    </div>
    <div class="spot sm act ">
    </div>
    <div class="spot sm foc act ">
    </div>
    <div class="spot sm dis ">
    </div>
    <div class="drag sm">
      drag
    </div>
    <div class="drag hov sm">
      drag
    </div>
    <div class="drag foc sm">
      drag
    </div>
    <div class="drag act sm">
      drag
    </div>
    <div class="drag foc act sm">
      drag
    </div>
    <div class="drag dis sm">
      drag
    </div>
    <div class="drop sm">
    </div>
    <div class="drop hov sm">
    </div>
    <div class="drop foc sm">
    </div>
    <div class="drop act sm">
    </div>
    <div class="drop foc act sm">
    </div>
    <div class="drop dis sm">
    </div>
    <div class="drop sm">
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop hov sm">
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop foc sm">
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop act sm">
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop foc act sm">
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop dis sm">
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop sm">
      With text
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop hov sm">
      With text
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop foc sm">
      With text
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop act sm">
      With text
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop foc act sm">
      With text
      <div class="drag sm">
        drag
      </div>
    </div>
    <div class="drop dis sm">
      With text
      <div class="drag sm">
        drag
      </div>
    </div>
    <input type="text"
           value="Button text"
           class="text sm"
    >
    <input type="text"
           value="Button text"
           class="text hov sm"
    >
    <input type="text"
           value="Button text"
           class="text foc sm"
    >
    <input type="text"
           value="Button text"
           class="text act sm"
    >
    <input type="text"
           value="Button text"
           class="text foc act sm"
    >
    <input type="text"
           value="Button text"
           class="text dis sm"
    >
    <select class="select sm">
      <option>
        Button text
      </option>
    </select>
    <select class="select hov sm">
      <option>
        Button text
      </option>
    </select>
    <select class="select foc sm">
      <option>
        Button text
      </option>
    </select>
    <select class="select act sm">
      <option>
        Button text
      </option>
    </select>
    <select class="select foc act sm">
      <option>
        Button text
      </option>
    </select>
    <select class="select dis sm">
      <option>
        Button text
      </option>
    </select>
  </div>
</div>
`;
